dpe/src/commands/certify_key.rs

```diff
@@ -150,7 +150,58 @@ impl CommandExecution for CertifyKeyCmd {
                 if !dpe.support.csr() {
                     return Err(DpeErrorCode::ArgumentNotSupported);
                 }
-                return Err(DpeErrorCode::ArgumentNotSupported);
+
+                let mut cert_req_info_buffer = [0u8; MAX_CERT_SIZE];
+                let mut cert_req_info_writer = X509CertWriter::new(&mut cert_req_info_buffer, true);
+                if issuer_len > MAX_CHUNK_SIZE {
+                    return Err(DpeErrorCode::InternalError);
+                }
+                let mut bytes_written = cert_req_info_writer.encode_certification_request_info(
+                    &pub_key,
+                    &subject_name,
+                    &measurements,
+                )?;
+                if bytes_written > MAX_CERT_SIZE {
+                    return Err(DpeErrorCode::InternalError);
+                }
+
+                let cert_req_info_digest = env
+                    .crypto
+                    .hash(
+                        DPE_PROFILE.alg_len(),
+                        &cert_req_info_buffer[..bytes_written],
+                    )
+                    .map_err(|_| DpeErrorCode::HashError)?;
+                let cert_req_info_sig = env
+                    .crypto
+                    .ecdsa_sign_with_alias(DPE_PROFILE.alg_len(), &cert_req_info_digest)
+                    .map_err(|_| DpeErrorCode::CryptoError)?;
```

Comment:
PKCS#10 CSRs should be self-signed (signed the the private key returned by `derive_key_pair`). This proves ownership of the key (this is also why we wrap the CSR in a CMS structure. So we can sign the whole thing with a different key).

Reply:
Done